File: src/main/java/Problem7.java
import utils.Euler;

public class Problem7 {
	public static void main(String[] args) {
		System.out.println(solution());
	}

	public static int solution() {
		int count = 1;
		for (int i = 3; true; i+=2) {
			if (Euler.isPrime(i)) 
				if (++count == 10001) 
					return i;
		}
	}
}

File: src/main/scala/Problem7.scala
package scala

import utils.Euler

object Problem7 extends App{
  def f(n:Int, a:Int):Int = if (Euler.isPrime(a)) if(n == 1) a else f(n-1, a+1) else f(n, a+1)
  print(f(10001, 2))
}
